wiki/Spotlight Guide.md

```diff
@@ -0,0 +1,142 @@
+
+    // Corresponds to the topic fragment's revision tab.
+    SpotlightViewState topic_revision_tab = 3;
+
+    // Add and describe your new spotlit feature here.
+    SpotlightViewState your_feature_name = <next_available_index>;
+}
+```
+
+And somewhere to store it (in [SpotlightStateDatabase](https://github.com/oppia/oppia-android/blob/d2c37dc547f3e5d12dfe62fa97b9b16fbf0fed6e/model/src/main/proto/spotlight.proto#L53)):
+```proto
+message SpotlightStateDatabase {
+  // Corresponds to the onboarding screen's next button.
+  SpotlightViewState onboarding_next_button = 1;
+
+  // Corresponds to the topic fragment's lessons tab.
+  SpotlightViewState topic_lesson_tab = 2;
+
+  // Corresponds to the topic fragment's revision tab.
+  SpotlightViewState topic_revision_tab = 3;
+
+  // Similarly, add storage for your new feature here.
+  SpotlightViewState your_feature_name = <next_available_index>;
+}
+```
+
+The new spotlight feature also needs to be added to the [SpotlightStateController](https://github.com/oppia/oppia-android/blob/d2c37dc547f3e5d12dfe62fa97b9b16fbf0fed6e/domain/src/main/java/org/oppia/android/domain/spotlight/SpotlightStateController.kt#L3). In the [retrieveSpotlightViewState](https://github.com/oppia/oppia-android/blob/d2c37dc547f3e5d12dfe62fa97b9b16fbf0fed6e/domain/src/main/java/org/oppia/android/domain/spotlight/SpotlightStateController.kt#L80) and the [recordSpotlightStateAsync](https://github.com/oppia/oppia-android/blob/d2c37dc547f3e5d12dfe62fa97b9b16fbf0fed6e/domain/src/main/java/org/oppia/android/domain/spotlight/SpotlightStateController.kt#L110) method, add the feature to the switch case as so:
+```kotlin
+fun retrieveSpotlightViewState(
+    profileId: ProfileId,
+    feature: Spotlight.FeatureCase,
+  ): DataProvider<SpotlightViewState> {
+    return retrieveCacheStore(profileId)
+      .transformAsync(
+        RETRIEVE_SPOTLIGHT_CHECKPOINT_DATA_PROVIDER_ID
+      ) {
+        val viewState = when (feature) {
+          ONBOARDING_NEXT_BUTTON -> it.onboardingNextButton
+          TOPIC_LESSON_TAB -> it.topicLessonTab
+          TOPIC_REVISION_TAB -> it.topicRevisionTab
+          YOUR_FEATURE_NAME -> it.yourFeatureName
+```
+
+Similarly for the [recordSpotlightStateAsync](https://github.com/oppia/oppia-android/blob/d2c37dc547f3e5d12dfe62fa97b9b16fbf0fed6e/domain/src/main/java/org/oppia/android/domain/spotlight/SpotlightStateController.kt#L110) method.
+
+## Hooking up the spotlight
+In order to request a spotlight in the UI, use [SpotlightManager](https://github.com/oppia/oppia-android/blob/d2c37dc547f3e5d12dfe62fa97b9b16fbf0fed6e/app/src/main/java/org/oppia/android/app/spotlight/SpotlightManager.kt#L4):
+
+```kotlin
+checkNotNull(getSpotlightManager()).requestSpotlight(
+	SpotlightTarget(
+	binding.yourFeatureName,
+	R.string.your_feature_string_string_resource,
+	SpotlightShape.RoundedRectangle,
+	Spotlight.FeatureCase.YOUR_FEATURE_NAME
+)
+)
+
+  private fun getSpotlightManager(): SpotlightManager? {
```

Comment:
Make sure the indentation of this matches checkNotNull() above

Reply:
done, PTAL